test/framework/type_checking.py

```diff
@@ -85,6 +85,42 @@ class Foo():
             pass
         self.assertErrorRegex(EasyBuildError, "No conversion function available", convert_value_type, None, Foo)
 
+    def test_to_toolchain(self):
+        """ Test toolchain string to dict conversion """
+        # normal cases
+        self.assertEqual(to_toolchain("intel, 2015a"), {'name': 'intel', 'version': '2015a'})
+        self.assertEqual(to_toolchain(['gcc', '4.7']), {'name': 'gcc', 'version': '4.7'})
+
+        # to tuple
+        self.assertEqual(to_toolchain("intel, 2015a", convert_to='tuple'), ('intel', '2015a'))
+        self.assertEqual(to_toolchain(['gcc', '4.7'], convert_to='tuple'), ('gcc', '4.7'))
+
+        # wrong type
+        self.assertErrorRegex(EasyBuildError, r"Conversion of .* \(type .*\) to toolchain dict is not supported",
+            to_toolchain, ('intel', '2015a'))
+
+        # wrong number of elements
+        errstr = "Can not convert .* to name and version .*. Expected 2 elements"
+        self.assertErrorRegex(EasyBuildError, errstr, to_toolchain, "intel, 2015, a")
+        self.assertErrorRegex(EasyBuildError, errstr, to_toolchain, "intel")
+        self.assertErrorRegex(EasyBuildError, errstr, to_toolchain, ['gcc', '4', '7'])
+
+    def to_dependency(self):
```

Comment:
you need to call this `test_to_dependency`, otherwise the test is not run (only stuff that starts with `test` is run as a test)